tiledb/sm/cpp_api/array.h

```diff
@@ -46,6 +46,128 @@
 
 namespace tiledb {
 
+/** Marker class to enforce a TimeTravel TemporalPolicy. */
+class TimeTravelMarker {};
+static constexpr TimeTravelMarker TimeTravel{};
+
+/** Marker class to enforce a RangeSelector TemporalPolicy. */
+class RangeSelectorMarker {};
```

Comment:
IMO `TimeIntervalMarker` or `TimeEndpointsMarker` or `TimeRangeMarker` might reduce any conflating association with "(query) range selection". Even though this is _practically_ an internal detail, it shows up very early in this public API file.

Reply:
(can we move the struct definition to the impl namespace?)

Reply:
I think `TimeRangeMarker` is an appropriate substitution. 

I'm not sure we want to move the new classes into `namespace impl` - the _only_ C++ Array constructor now uses these classes. We may not want to complicate usage for users. I could misunderstand the intentions of `impl` (or your change request), but I think it'll unnecessarily complicate things to require `...impl::TemporalPolicy(impl::TimeTravel, timestamp), impl::TimeRange(impl::AESGCM, key))`

Reply:
> I could misunderstand the intentions of impl (or your change request), but I think it'll unnecessarily complicate things to require

Sorry, what I meant here was:
- declare `struct TimeTravelMarker {}` and `struct TimeRangeMarker {}` in `impl`
- keep the the constexpr definitions here, since that's the user-facing part of the API
```
static constexpr impl::TimeRangeMarker TimeRangeSelector{};
```
But feel free to disregard.